cpp/src/arrow/dataset/discovery.h

```diff
@@ -85,6 +85,7 @@ class ARROW_DS_EXPORT DataSourceDiscovery {
   ExpressionPtr root_partition_;
 };
 
+// TODO(kszucs): a static Defaults method would be nice
```

Comment:
@pitrou why do we use a default static method in other options struct?

Reply:
Just being explicit :-)

Reply:
I like the defaults static method BTW, rust has a trait for that which provides a pretty and explicit API.